src/main/scala/io/druid/indexer/spark/SparkDruidIndexer.scala

```diff
@@ -86,7 +86,16 @@ object SparkDruidIndexer {
       s => {
         val p = new Path(s)
         val fs = p.getFileSystem(sc.hadoopConfiguration)
-        fs.getFileStatus(p).getLen
+        if (fs.getFileStatus(p).isDirectory) {
+          var sum = 0l
+          val children = fs.listFiles(p, false)
```

Comment:
I'd prefer for Hadoop to do things in a way that is compatible with Scala :-P

barring that, please add a comment to why the manual iteration is needed.